packages/fxa-settings/src/components/CardHeader/index.tsx

```diff
@@ -4,41 +4,89 @@
 
 import React from 'react';
 import { FtlMsg } from 'fxa-react/lib/utils';
+import { MozServices } from '../../lib/types';
 
-// TODO improve typing to allow either default or custom or both (but not none)
-export type CardHeaderProps = {
-  headingWithDefaultServiceFtlId: string;
-  headingWithCustomServiceFtlId?: string;
+interface CardHeaderRequiredProps {
   headingText: string;
-  serviceName?: string;
-};
+}
+
+interface CardHeaderDefaultServiceProps extends CardHeaderRequiredProps {
+  headingWithDefaultServiceFtlId: string;
+}
+
+interface CardHeaderCustomServiceProps extends CardHeaderRequiredProps {
+  headingWithCustomServiceFtlId: string;
+  serviceName: Exclude<MozServices, 'Default'>;
+}
+
+interface CardHeaderBasicProps extends CardHeaderRequiredProps {
+  headingTextFtlId: string;
+}
+
+export type CardHeaderProps =
+  | CardHeaderDefaultServiceProps
+  | CardHeaderCustomServiceProps
+  | CardHeaderBasicProps;
+
+function isCustomService(
+  props: CardHeaderProps
```

Comment:
@vpomerleau @millsoper 

I noticed we had `card-header` `h1s` scattered about, and it felt like all variations should go in this component. I'm going to file a 1-pointer here to update all other instances of using `card-header` to use this instead so our margins are consistent as well but tagging you both as an FYI so we can just start using this instead.

I also removed the service versions' `<header>` wrapper. I feel like we didn't really need it, since we don't have a `footer` that corresponds with it, which is typical when using that elem. 🤷‍♀️ 

This was the nicest way I could find to do this but I'm open to anything better.

Reply:
Nice, great thinking w the ticket -- and it's great to see those `<header>` elems getting cleaned up!

Reply:
Hmmm...I find these checks a little hard to read, probably due to less familiarity with Typescript, but having to learn more Typescript is certainly not a bad thing 😂

Reply:
@millsoper They look a little funky if you haven't seen [type predicates / guards](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) much, I copied straight from their docs. I might have been able to use a different approach but this is one I've used before and one we have in `ApolloErrorAlert` (which I also wrote). Definitely open to any suggestions or future modifications!